master/docs/relnotes/index.rst

```diff
@@ -34,6 +34,10 @@ Deprecations, Removals, and Non-Compatible Changes
 Changes for Developers
 ~~~~~~~~~~~~~~~~~~~~~~
 
+* :class:`~buildbot.steps.source.base.Source` now has a ``doWithRetry`` method to help make it easier
```

Comment:
why only for sources? wouldn't be better in a mixin or something that can apply everywhere?